src/lib.rs

```diff
@@ -155,15 +158,27 @@ pub fn run(mut vm: VMState) -> (U256, VMState) {
                 // TODO: This is not how return works. Fix when we have calls between contracts
                 // hooked up.
                 // This is only to keep the context for tests
-                Variant::Ret(_) => {
-                    if vm.running_contexts.len() > 1
-                        || !vm.current_context().near_call_frames.is_empty()
-                    {
-                        vm.pop_frame()
-                    } else {
-                        break;
+                Variant::Ret(ret_variant) => match ret_variant {
+                    RetOpcode::Ok => {
+                        let should_break = _ok(&mut vm, &opcode);
+                        if should_break {
+                            break;
+                        }
                     }
-                }
+                    RetOpcode::Revert => {
+                        let should_break = _revert(&mut vm, &opcode);
+                        if should_break {
+                            panic!("Contract Reverted");
```

Comment:
Since we're implementing the panic opcode, should still be a panic?
Maybe a Revert should be reported as an Err instead of panicking.

Reply:
I left it as a panic in order to continue the same approach, I agree with you, this is being implemented on https://github.com/lambdaclass/era_vm/issues/45 , if you are ok, we can leave this one, and focus on error handling on that issue, since its a big change